projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -185,13 +185,76 @@ public function build( $id ) {
 
 			// if task can't be found don't add anything
 			if ( ! empty( $task ) ) {
-				$tasks_for_task_list[] = $task;
+				$tasks_for_task_list[] = $this->build_task( $task );
 			}
 		}
 
 		return $tasks_for_task_list;
 	}
 
+	/**
+	 * Builds a single task with current state
+	 *
+	 * @param Task $task Task definition.
+	 * @return Task Task with current state.
+	 */
+	private function build_task( $task ) {
+		$built_task                 = array(
+			'id'    => $task['id'],
+			'title' => $task['title'],
+		);
+		$built_task['completed']    = $this->is_task_complete( $task );
+		$built_task['disabled']     = $this->is_task_disabled( $task );
+		$built_task['subtitle']     = $this->load_subtitle( $task );
+		$built_task['isLaunchTask'] = isset( $task['isLaunchTask'] ) ? $task['isLaunchTask'] : false;
+		$built_task['badge_text']   = isset( $task['badge_text'] ) ? $task['badge_text'] : '';
+		return $built_task;
+	}
+
+	/**
+	 * Loads a subtitle for a task, calling the callback if it exists.
+	 *
+	 * @param Task $task A task definition.
+	 * @return string The subtitle for the task.
+	 */
+	private function load_subtitle( $task ) {
+		if ( isset( $task['subtitle'] ) ) {
+			if ( is_callable( $task['subtitle'] ) ) {
+				return call_user_func( $task['subtitle'] );
+			}
+			return $task['subtitle'];
+		}
+		return '';
+	}
+
+	/**
+	 * Checks if a task is disabled
+	 *
+	 * @param array $task Task definition.
+	 * @return boolean
+	 */
+	public function is_task_disabled( $task ) {
+		if ( isset( $task['is_disabled'] ) && is_callable( $task['is_disabled'] ) ) {
+			return call_user_func( $task['is_disabled'] );
+		}
+		return false;
+	}
+
+	/**
+	 * Checks if a task is complete, relying on task-defined callbacks if available
+	 *
+	 * @param Task $task Task definition.
+	 * @return boolean
+	 */
+	public function is_task_complete( $task ) {
+		if ( isset( $task['is_complete_callback'] ) && is_callable( $task['is_complete_callback'] ) ) {
+			return call_user_func( $task['is_complete_callback'] );
```

Comment:
`NB:`

Does PHP not allow for invocation of the anonymous function like so?

`$task['is_complete_callback']( )`

Reply:
It would fail in the cases where we supply string or array-type callback identifiers like
`'some_function_callback'` or `array( $object, 'method_callback' )`. `call_user_func` allows us leverage any of the three formats (1. anonymous functions, 2. string identifier, 3. array identifier for object->method).

Reply:
> string identifier,  array identifier for object->method)

Ah these are what I was missing. Makes sense 👍

Reply:
The last two were formerly the only way to do callbacks in PHP, before anonymous functions landed in PHP 5.3